### ISSUE TITLE
 Bug: Review front end watching.

### ISSUE THREAD

**Issue by ajsantander** [aragon/aragon-cli#1216]
Atm, front + back end watching in the Buidler plugin is working, but the frontend part is in a pretty ugly state:
* `npm run build` is called on app/ before watching. Why is this needed? Can't we just watch?
* `npm run build` internally in app/ calls `npm run build:app` and `npm run build:script`. Why two builds?
* Building or watching is throwing some "npm update" errors.
* `npm run watch` specifies the src/script.js input. Why?
* `npm run watch` fires up a devserver when we are already firing up a liveserver in the plugin. Careful tho, does front + back hot reloading work if the app's devserver is removed?
* What the heck is `npm run sync-assets` and why do we need it?

Also, we need to consider what to do if the app doesn't have a watcher. Should we re-trigger build from the plugin?

TL;DR the process is super complicated and obscure right now, and we need to make it as clear as water, of course, while maintaining the best and fastest possible hot reloading for both front and back end.

**Comment by dapplion** [comment]
On our side:
 - I've assumed that the app has a watcher script, therefore, the first is not necessary
 - The `watch` script fires up a static web server (in this case by parcel) so it's not necessary to run a live server on our end.

Generalizing this won't be easy, so I would focus now on making the entire plugin work and then we'll try to find a way to generalize it to apps with parcel, webpack, or raw.

On the `aragon-react-boilerplate` side:
 - The app has two entry points, which seem to require different configuration, therefore, it may be required to have two scripts
  - `index.html`: which is built with `build:app`
  - `script.js`: which is built with `build:script`
 - The script `sync-assets` calls another Aragon NPM package `copy-aragon-ui-assets` which does some packaging or assets.

We would need to sync with Brett to get a better knowledge of why this was set up this way but in my opinion, generalizing this is out of the scope of sprint v1

**Comment by dapplion** [comment]
After some research, `parcel` support to `serve` with multiple entry points with
```
node_modules/.bin/parcel serve index.html src/script.js -p 8001 --out-dir ../dist/ --no-cache
```
however, the directory structure is respected so it looks like
```
dist/index.html
dist/src/script.js
```
where the `manifest.json` has to be changed to make it work.

-----

With this setup, if you change `script.js` the page inside the `iframe` is refreshed but the script is **not** re-run. Note that the Aragon client runs the script and the iframe in different sandboxes so refresh of the iframe does not affect the `script.js`
```js
const worker = new SandboxedWorker(scriptUrl, { name: workerName })
```
https://github.com/aragon/aragon/blob/e92dfb9c36d692edfb9e9dd6beab675a6680e5fd/src/aragonjs-wrapper.js#L228

-----

@sohkai Is there a way to trigger a refresh / re-run of the script's sandbox?
